gammapy/datasets/map.py

```diff
@@ -2778,14 +2779,16 @@ def _compute_npred(self):
         if isinstance(self.model, TemplateNPredModel):
             npred = self.model.evaluate()
         else:
-            npred = self.compute_flux_psf_convolved()
-
-            if self.model.apply_irf["exposure"]:
-                npred = self.apply_exposure(npred)
-
-            if self.model.apply_irf["edisp"]:
-                npred = self.apply_edisp(npred)
-
+            norm_only_changed, renorm = self.norm_only_changed()
+            if not (norm_only_changed and self.use_cache):
+                self._cached_parameter_previous = self.model.parameters.value
+                for method in self.methods_sequence:
+                    if len(inspect.signature(method).parameters) == 0:
```

Comment:
I would just remove the check here and introduce optional "dummy" arguments (like `*args`) to the methods that don't require any arguments